files/default/plugins/printers.rb

```diff
@@ -0,0 +1,44 @@
+provides 'ohai_gecos'
+
+if ohai_gecos.nil?
+  ohai_gecos Mash.new
+end
+
+begin
+  entries = []
+  printers = []
+
+  entries = Mixlib::ShellOut.new("lpstat -a | awk {'print $1'}")
+  entries.run_command
+
+  if entries.exitstatus == 0
+
+    entries.stdout.split(/\r?\n/).each do |printer|
+      lpoptions = Mixlib::ShellOut.new("lpoptions -p #{printer}")
+      lpoptions.run_command
+      lpq = Mixlib::ShellOut.new("lpq -P #{printer}")
+      lpq.run_command
+      lpstat = Mixlib::ShellOut.new("lpstat -p #{printer}")
+      lpstat.run_command
+
+      /printer-make-and-model='(?<printerBrand>.*)'/  =~ lpoptions.stdout
+      /printer-make-and-model='(?<printerDriver>.*)'/ =~ lpoptions.stdout
+      /device-uri=(?<printerConn>\S+)/                =~ lpoptions.stdout
+      printerStatus = lpstat.stdout
+      printerQueue  = lpq.stdout
+
+      thisprinter Mash.new
```

Comment:
No, it shouldn't because, in case there's more than one printer, all the information about it will be stored in thisprinter object and it must be create new in every iteration.